doc/pages/user/035-register_device.md

```diff
@@ -0,0 +1,54 @@
+
+#### "Local" or "Plant" Agents
+
+In this scenario, devices are imprinted with their *Credentials Secret* in the production plant. The Device might not even be connected to the Internet, whereas the machine running the Agent has access to the target Astarte Cluster and adequate Credentials for Registration.
+
+Once the Agent acquires the Device ID of the Device which should be registered, it issues the request to Astarte's Pairing API and obtains the Device's *Credentials Secret*. At this stage, the Agent is in charge of delivering the *Credentials Secret* to the Device the way it sees fit. As a best practice, the *Credentials Secret* should then be saved to an OTP area or a dedicated secure storage in the device to prevent tampering or accidental loss.
+
+Even though this is arguably the most secure mechanism available for Registering a Device, it might not fit every use case as the Device will be irrevocabily assigned to a specific Astarte Cluster and a specific Realm in that Cluster before it even connects.
+
+#### "Remote" Agents
+
+If your use case demands more flexibility, Registering a Device in a plant might not fit your Device's lifecycle. This could be likely if, for example, Realm or Cluster assignment should be done dynamically once the Device reaches its final user.
+
+In this case, this role is usually delegated to an external web application acting as an Agent. In this case, it's up to the user setting up all mechanisms for delivering the *Credentials Secret* to the Device, which includes securing the communication channel. On the other hand, this allows an extremely flexible approach to Registration, which can be implemented through an entirely custom logic.
+
+## Credentials Secret Lifecycle
+
+*Credentials Secrets* are meant to be immutable - as such, they should be handled with extreme care. *Credentials Secrets* are used only for interacting with Pairing, hence to obtain Credentials for a Transport which, on the other hand, are meant to be volatile.
+
+A Device can be Registered an arbitrary number of times before its *Credentials Secret* is used for the first time for interacting with Pairing. This is done to ensure the entire Registration process, including any kind of external custom logic of the Agents, has been carried over successfully, allowing a de-facto "retry" until there's certainty the Device has access to its *Credentials Secret*. Please note that when Registering a Device, a new *Credentials Secret* is generated every time.
+
+Once the *Credentials Secret* is used, Astarte prevents further registration of the same Device again. There's no defined procedure for substituting a *Credentials Secret* - it can be done by performing manual operation, but it should be considered an unusual/emergency procedure (e.g.: a Device has been tampered and got back to its plant with its previous *Credentials Secret* compromised).
```

Comment:
"Once the *Credentials Secret* is used," -> "Once the *Credentials Secret* is used to retrieve Credentials for a Transport for the first time,"